R/lump.R

```diff
@@ -56,46 +58,33 @@
 #'
 fct_lump <- function(f, n, prop, w = NULL, other_level = "Other",
                      ties.method = c("min", "average", "first", "last", "random", "max")) {
-  f <- check_factor(f)
-  w <- check_weights(w, length(f))
-  ties.method <- match.arg(ties.method)
 
+  ties.method <- match.arg(ties.method)
+  calcs <- calc_levels(f, w)
```

Comment:
I think it would be clearer to keep the variables in `calcs` (i.e. don't unpack them)